plasma_framework/contracts/src/transactions/PaymentTransactionModel.sol

```diff
@@ -24,8 +24,24 @@ library PaymentTransactionModel {
         bytes32 metaData;
     }
 
-    function decode(bytes memory _tx) internal pure returns (PaymentTransactionModel.Transaction memory) {
-        RLPReader.RLPItem[] memory rlpTx = _tx.toRlpItem().toList();
+    /*
+     * @notice Decodes a encoded byte array into a PaymentTransaction
+     * The following rules about the rlp-encoded transaction are enforced:
+     *      - `txType` must be an integer value with no leading zeros
+     *      - `inputs` is an list of 0 to 4 elements
+     *      - Each `input` is a 32 byte long array
+     *      - An `input` may not be all zeros
+     *      - `outputs` is an list of 0 to 4 elements
+     *      - `output.outputType` must be an integer value with no leading zeros
+     *      - `output.outputGuard` is a 20 byte long array
+     *      - `output.token` is a 20 byte long array
+     *      - `output.amount` must be an integer value with no leading zeros
+     *      - An `output` may not be null; A null output is one whose amount is zero
+     * @param tx An RLP-encoded transaction
+     * @return A decoded PaymentTransaction struct
+     */
+    function decode(bytes memory tx) internal pure returns (PaymentTransactionModel.Transaction memory) {
```

Comment:
my IDE argues about the `tx` variable

> This declaration shadows a builtin symbol.

Reply:
Nice catch, I wonder why sol linter is picking these up?